packages/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/network/DevLauncherNetworkLogger.kt

```diff
@@ -80,19 +80,33 @@ class DevLauncherNetworkLogger private constructor() {
         ))
       }
     }
-    val data = JSONObject(mapOf(
+    val requestWillBeSentData = JSONObject(mapOf(
       "method" to "Network.requestWillBeSent",
-      "params" to params,
+      "params" to requestWillBeSentParams,
     ))
-    inspectorPackagerConnection.sendWrappedEventToAllPages(data.toString())
+    inspectorPackagerConnection.sendWrappedEventToAllPages(requestWillBeSentData.toString())
+
+    val extraInfoParams = mapOf(
+      "requestId" to requestId,
+      "associatedCookies" to emptyList<Void>(),
+      "headers" to request.headers().toSingleMap(),
+      "connectTiming" to mapOf(
+        "requestTime" to now,
+      ),
+    )
+    val extraInfodata = JSONObject(mapOf(
+      "method" to "Network.requestWillBeSentExtraInfo",
+      "params" to extraInfoParams
+    ))
+    inspectorPackagerConnection.sendWrappedEventToAllPages(extraInfodata.toString())
   }
 
   /**
    * Emits CDP `Network.responseReceived` and `Network.loadingFinished` events
    */
   fun emitNetworkResponse(request: Request, requestId: String, response: Response) {
     val now = BigDecimal(System.currentTimeMillis() / 1000.0).setScale(3, RoundingMode.CEILING)
-    var params = mapOf(
+    val responseReceivedParams = mapOf(
       "requestId" to requestId,
       "loaderId" to "",
       "hasExtraInfo" to false,
```

Comment:
do you think we should set the `hasExtraInfo` as true when we send the `Network.requestWillBeSentExtraInfo`?

and also [`redirectHasExtraInfo`](https://chromedevtools.github.io/devtools-protocol/tot/Network/#event-requestWillBeSent) maybe

Reply:
Yep, you are right! 
<img width="780" alt="image" src="https://user-images.githubusercontent.com/1203991/229622766-039ae3e8-26f4-4dcc-bad5-d7a56dc44cc8.png">

I assume that the redirect request is triggered with the normal events as is (not something we have to implement manually, right?)

Reply:
```suggestion
      "hasExtraInfo" to true,
```

Reply:
I think you are right about the `redirectHasExtraInfo`:

<img width="780" alt="image" src="https://user-images.githubusercontent.com/1203991/229622766-039ae3e8-26f4-4dcc-bad5-d7a56dc44cc8.png">

But I don't think we have to enable the `hasExtraInfo` on [`Network.responseReceived`](https://chromedevtools.github.io/devtools-protocol/tot/Network/#event-responseReceived), because that refers to the [`Network.responseReceivedExtraInfo`](https://chromedevtools.github.io/devtools-protocol/tot/Network/#event-responseReceivedExtraInfo).

So far, this change is only required because of the visual "Provisional headers are shown" issue. Which seems to be an oversight in the spec itself, there is no option to say "we are not implementing the `Network.requestWillBeSentExtraInfo` event". Luckily, we can do that with the `hasExtraInfo`, and we have disabled it.

Reply:
makes sense and nice explanation! thanks